### PR TITLE
parser: remove YYParse function, use Parse and ParseOneStmt instead.

### PR DESCRIPTION
**Pull request by coocood**

### CODE REVIEW

File: parser/yy_parser.go [hunk 1]
```diff
@@ -25,7 +30,33 @@ const (
...
+	yyParse(l)
+	if len(l.Errors()) != 0 {
+		return nil, errors.Trace(l.Errors()[0])
+	}
+	return l.Stmts(), nil
+}
+
+// ParseOne parses a query and return the ast.StmtNode.
+// The query must has one statement, otherwise ErrSyntax is returned.
+func ParseOne(sql, charset, collation string) (ast.StmtNode, error) {
```
Comment (qiuyesuifeng) [inline parser/yy_parser.go]: How about ParseOneStmt?

File: parser/yy_parser.go [hunk 2]
```diff
@@ -25,7 +30,33 @@ const (
...
+func Parse(sql, charset, collation string) ([]ast.StmtNode, error) {
+	if charset == "" {
+		charset = mysql.DefaultCharset
+	}
+	if collation == "" {
+		collation = mysql.DefaultCollationName
+	}
+	l := NewLexer(sql)
+	l.SetCharsetInfo(charset, collation)
+	yyParse(l)
```
Comment (qiuyesuifeng) [inline parser/yy_parser.go]: Should we handle the return value?

Comment (coocood) [inline parser/yy_parser.go]: We have `Errors()` method to check error, which is enough, the integer return code is redundant.

File: parser/yy_parser.go [hunk 3]
```diff
@@ -25,7 +30,33 @@ const (
...
+	if len(l.Errors()) != 0 {
+		return nil, errors.Trace(l.Errors()[0])
+	}
+	return l.Stmts(), nil
+}
+
+// ParseOneStmt parses a query and return the ast.StmtNode.
```
Comment (shenli) [inline parser/yy_parser.go]: returns a ast.StmtNode

File: parser/yy_parser.go [hunk 4]
```diff
@@ -25,7 +30,33 @@ const (
...
+// The query must has one statement, otherwise ErrSyntax is returned.
```
Comment (shenli) [inline parser/yy_parser.go]: must have

File: parser/yy_parser.go [hunk 5]
```diff
@@ -25,7 +30,33 @@ const (
 	CodeSyntaxErr terror.ErrCode = iota + 1
 )
 
-// YYParse is an wrapper of `yyParse` to make it exported.
-func YYParse(yylex yyLexer) int {
-	return yyParse(yylex)
+// Parse parses a query string to raw ast.StmtNode.
+// If charset and collation is "", default charset and collation will be used.
```
Comment (shenli) [inline parser/yy_parser.go]: and->or

**Comment by shenli** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**